File: bukkit/elysium-chat-bukkit/src/main/java/com/seventh_root/elysium/chat/bukkit/chatchannel/pipeline/BukkitLogChatChannelPipelineComponent.java
package com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline;

import com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent;
import com.seventh_root.elysium.api.chat.ChatMessageContext;

import java.io.IOException;

import static com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent.Type.POST_PROCESSOR;

public class BukkitLogChatChannelPipelineComponent extends ChatChannelPipelineComponent {

    @Override
    public Type getType() {
        return POST_PROCESSOR;
    }

    @Override
    public String process(String message, ChatMessageContext context) {
        try {
            context.getChatChannel().log(message);
        } catch (IOException exception) {
            exception.printStackTrace();
        }
        return message;
    }

}

File: bukkit/elysium-chat-bukkit/src/main/kotlin/com/seventh_root/elysium/chat/bukkit/chatchannel/pipeline/BukkitLogChatChannelPipelineComponent.kt
package com.seventh_root.elysium.chat.bukkit.chatchannel.pipeline

import com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent
import com.seventh_root.elysium.api.chat.ChatChannelPipelineComponent.Type.POST_PROCESSOR
import com.seventh_root.elysium.api.chat.ChatMessageContext
import com.seventh_root.elysium.api.chat.ChatMessagePostProcessContext
import java.io.IOException

class BukkitLogChatChannelPipelineComponent : ChatChannelPipelineComponent() {

    override val type: ChatChannelPipelineComponent.Type
        get() = POST_PROCESSOR

    override fun process(message: String, context: ChatMessageContext): String {
        return message
    }

    override fun postProcess(message: String, context: ChatMessagePostProcessContext): String {
        try {
            context.chatChannel.log(message)
        } catch (exception: IOException) {
            exception.printStackTrace()
        }
        return message
    }

}
